.github/workflows/cypress.yml

```diff
@@ -0,0 +1,156 @@
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Download MrBeamPlugin image
+        uses: actions/download-artifact@v3
+        with:
+          name: mrbeam_plugin
+          path: /tmp
+      - name: Load Docker image
+        run: |
+          apt-get --allow-releaseinfo-change update
+          apt-get install docker.io -y
+          docker load --input /tmp/image.tar
+          docker run --name mrbeam-plugin -d -p5002:5000 mrbeam_plugin
+      - run: |
+          envsubst < ./cypress/fixtures/test-data.template.json > ./cypress/fixtures/test-data.json
+          envsubst < ./cypress.config-template.js > ./cypress.config.js
+      - name: Run Cypress e2e - Chrome
+        uses: cypress-io/github-action@v4
+        with:
+          wait-on: ${{ env.MRBEAM_PLUGIN_URL }}
+          browser: chrome
+          config-file: cypress.config.js
+          command: yarn ts-node cypress/scripts/cypress-ci-run.ts
+        env:
+          TOTAL_RUNNERS: 5
+          THIS_RUNNER: ${{ matrix.containers }}
+          # Recommended: pass the GitHub token lets this action correctly
+          # determine the unique run id necessary to re-run the checks
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Upload JUnit reports
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: report_${{ matrix.containers }}
+          path: cypress/results
+
+  xray-import:
+    runs-on: ubuntu-latest
+    needs: ui-chrome-tests
+    if: always()
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+      - name: Download JUnit reports
+        uses: actions/download-artifact@v3
+        with:
+          path: cypress/results
+      - name: Display structure of downloaded files
+        run: ls -R
+        working-directory: cypress/results
+      - name: Merge reports
+        run: |
+          npm install --only=dev
+          npm run merge-reports
+      - name: Import to Xray
+        id: xray
+        run: |
+          token=$(curl -H "Content-Type: application/json" -X POST --data '{ "client_id": "${{ secrets.JIRA_CLIENT_ID }}","client_secret": "${{ secrets.JIRA_CLIENT_SECRET }}" }'  https://xray.cloud.getxray.app/api/v2/authenticate| tr -d '"')
+          response=$(curl -H "Content-Type: text/xml" -X POST -H "Authorization: Bearer $token"  --data @"cypress/results/combined.xml" https://xray.cloud.getxray.app/api/v2/import/execution/junit?projectKey=SW)
+          echo "response=$response" >> $GITHUB_OUTPUT
+      - name: Change issue summary and add link
+        # issue_url: extracts the newly created test execution issue url from the previous response
+        # issue_name: contains the latest commit message. Should be something like "SW-1234 summary"
+        # tests_issue: extracts the "SW-1234" from the commit message, to add it as a link to the text execution issue
```

Comment:
What happens if the last commit message doesn't have the jira issue? and also even better... what should it happen? Maybe fail the build?

Reply:
It will fail, but because linking the issue to the test plan will fail. Then the build will fail, but the previous step would have succeeded (which is importing the results to Xray)